From 99ffe9e8feaedee9abc58427e06d1f17000ce827 Mon Sep 17 00:00:00 2001
From: Ian Campbell <ijc@hellion.org.uk>
Date: Mon, 20 Dec 2010 11:31:20 +0000
Subject: [PATCH] drm/ttm: Add ttm_tt_free_page

Simplifies backport of
git://git.kernel.org/pub/scm/linux/kernel/git/konrad/xen.git#devel/ttm.pci-api-v2
in the absence of 1403b1a38e8b "drm/ttm: add pool wc/uc page allocator V3".

Signed-off-by: Ian Campbell <ian.campbell@citrix.com>
---
 drivers/gpu/drm/ttm/ttm_tt.c |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/drivers/gpu/drm/ttm/ttm_tt.c b/drivers/gpu/drm/ttm/ttm_tt.c
index a759170..48ec9aa 100644
--- a/drivers/gpu/drm/ttm/ttm_tt.c
+++ b/drivers/gpu/drm/ttm/ttm_tt.c
@@ -87,6 +87,11 @@ static struct page *ttm_tt_alloc_page(unsigned page_flags)
 	return alloc_page(gfp_flags);
 }
 
+static void ttm_tt_free_page(struct page *page, unsigned page_flags)
+{
+	__free_page(page);
+}
+
 static void ttm_tt_free_user_pages(struct ttm_tt *ttm)
 {
 	int write;
@@ -314,7 +319,7 @@ static void ttm_tt_free_alloced_pages(struct ttm_tt *ttm)
 				       "Leaking pages.\n");
 			ttm_mem_global_free_page(ttm->glob->mem_glob,
 						 cur_page);
-			__free_page(cur_page);
+			ttm_tt_free_page(cur_page, ttm->page_flags);
 		}
 	}
 	ttm->state = tt_unpopulated;
-- 
1.5.6.5

